src/services/dataSource.js

```diff
@@ -0,0 +1,43 @@
+export const getDataSources = (appId) => {
+  const getOptions = appId ? {appId: appId} : {};
+
+  return Fliplet.DataSources.get(getOptions);
+};
+
+export const getDataSource = (dataSourceId) => {
+  return Fliplet.DataSources.getById(dataSourceId);
+};
+
+export const createDataSource = (widgetData) => {
+  return Fliplet.Modal.prompt({
+    title: 'Enter a name for the data source',
+    value: widgetData.default.name
+  })
+    .then(dataSourceName => {
+      if (dataSourceName === null) {
+        return false;
```

Comment:
Does it need to return `false`?

```suggestion
        return;
```

Reply:
@tonytlwu No, it doesn't. I'll remove it.